java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -338,18 +387,22 @@
  */
 JNIEXPORT jstring JNICALL Java_ai_onnxruntime_OnnxTensor_getString
   (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle, jlong allocatorHandle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
+    (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
     const OrtApi* api = (const OrtApi*) apiHandle;
     // Extract a String array - if this becomes a performance issue we'll refactor later.
-    jobjectArray outputArray = createStringArrayFromTensor(jniEnv,api, (OrtAllocator*) allocatorHandle, (OrtValue*) handle);
-
-    // Get reference to the string
-    jobject output = (*jniEnv)->GetObjectArrayElement(jniEnv, outputArray, 0);
+    jobjectArray outputArray = createStringArrayFromTensor(jniEnv, api, (OrtAllocator*) allocatorHandle,
```

Comment:
>createStringArrayFromTensor [](http://example.com/codeflow?start=31&length=27)

My understanding this is a single string case. Does the function check it is a single element tensor?

Reply:
We check that in Java before this call. The individual "getX" methods are private and only called if it's a single element.